src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -7,18 +7,32 @@
 using Microsoft.AspNet.JsonPatch.Exceptions;
 using Microsoft.AspNet.JsonPatch.Helpers;
 using Microsoft.AspNet.JsonPatch.Operations;
-using Newtonsoft.Json;
 using Newtonsoft.Json.Serialization;
 
 namespace Microsoft.AspNet.JsonPatch.Adapters
 {
+    /// <inheritdoc />
     public class ObjectAdapter<T> : IObjectAdapter<T> where T : class
     {
+        /// <summary>
+        /// Gets or sets the <see cref="IContractResolver"/>.
+        /// </summary>
         public IContractResolver ContractResolver { get; set; }
 
-        public ObjectAdapter(IContractResolver contractResolver)
+        /// <summary>
+        /// Action for logging <see cref="JsonPatchError{T}"/>.
+        /// </summary>
+        public Action<JsonPatchError<T>> LogErrorAction { get; private set; }
```

Comment:
If you want to you can leave out the `private set` and earn PranavPoints to spend on cool prizes. Also properties go after constructors. Since you're doing other cleanup here, might as well do that too.